common/flatpak-installation.c

```diff
@@ -1655,6 +1655,31 @@ flatpak_installation_get_config (FlatpakInstallation *self,
   return flatpak_dir_get_config (dir, key, error);
 }
 
+/**
+ * flatpak_installation_get_default_locales:
+ * @self: a #FlatpakInstallation
+ * @error: return location for a #GError
+ *
+ * Get the default languages and locales used by frontends like
+ * GNOME Software to decide which language-specific apps to display.
+ *
+ * Returns: (array zero-terminated=1) (element-type utf8) (transfer full):
+ *   A possibly empty array of language and locale strings, or %NULL on error.
+ * Since: 1.5.1
+ */
+char **
+flatpak_installation_get_default_locales (FlatpakInstallation  *self,
+                                          GError              **error)
+{
+  g_autoptr(FlatpakDir) dir = NULL;
+
+  dir = flatpak_installation_get_dir (self, error);
+  if (dir == NULL)
+    return NULL;
+
+  return flatpak_dir_get_default_locales (dir);
```

Comment:
Is there a reason we don't want this to behave like `flatpak_installation_get_default_languages()` with respect to `xa.languages`? If `xa.languages` is set it will be used for locale extensions, so I think it should be returned here too.

Reply:
Oh, I haven't considered that. I should make a counter-part of the `flatpak_dir_get_locale_languages` which uses the `flatpak_dir_get_default_locales` instead. I hope the name flatpak_dir_get_locales is not very confusing, since if the xa.languages is set, it won't be returning any locales :/

Reply:
I think it's alright. If `xa.extra-languages` is set to just languages it won't return any locales in that case either. I think the term locales is a superset of the term languages in this context.